### PR TITLE
Judicial system/Defender

### PR DESCRIPTION
**Pull request by oddsson**
# Defender name and email to step one

https://app.asana.com/0/1197628843274291/1198949971788356

## What

Add fields for defender email and name on step one

## Why

It's a required.

## Screenshots / Gifs

screencapture-localhost-4200-krafa-131f17d3-7593-4c63-8c69-d8c58faea8b6-2020-11-06-15_16_06

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

**Review by addi — DISMISSED** [review]
LGTM

File: apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.spec.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ import { MockedProvider } from '@apollo/client/testing'
 import * as Constants from '../../../utils/constants'
 
 describe('/domari-krafa/fyrirtokutimi', () => {
-  test('should not allow users to continue unless every required field has been filled out', async () => {
+  test('should not allow users to continue unless every required field has been filled out rass', async () => {
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.spec.tsx:17]:
```suggestion
  test('should not allow users to continue unless every required field has been filled out', async () => {
```

Comment (oddsson) [reply apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.spec.tsx:17]: 😅

File: apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.spec.tsx [hunk 2]
```diff
@@ -68,7 +76,7 @@ describe('/domari-krafa/fyrirtokutimi', () => {
     ).not.toBeDisabled()
   })
 
-  test('should have a prefilled court date with requested court date', async () => {
+  test('should have a prefilled court date and dedender info with requested info', async () => {
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.spec.tsx:79]: Það ætti að gera bæði, prefill-a með court date og defender info

File: apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.spec.tsx [hunk 3]
```diff
@@ -34,6 +34,14 @@ describe('/domari-krafa/fyrirtokutimi', () => {
               id: 'test_id_2',
               courtRoom: '999',
             } as UpdateCase,
+            {
+              id: 'test_id_2',
+              defenderName: 'rassi',
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.spec.tsx:39]: Getum við plís haft eitthvað annað en rassi og rassi@prump.is? 🙏

File: apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.tsx [hunk 4]
```diff
@@ -85,6 +85,28 @@ export const HearingArrangements: React.FC = () => {
         )
       }
 
+      console.log('bla: ', theCase.defenderName)
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.tsx:88]:
```suggestion
```

File: apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.tsx [hunk 5]
```diff
@@ -85,6 +85,28 @@ export const HearingArrangements: React.FC = () => {
...
+
+      if (!theCase.defenderName) {
+        theCase = { ...theCase, defenderName: theCase.requestedDefenderName }
+
+        console.log('rassi was here')
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Judge/HearingArrangements/HearingArrangements.tsx:93]:
```suggestion
```

File: apps/judicial-system/web/src/utils/mocks.ts [hunk 6]
```diff
@@ -117,6 +117,8 @@ const testCase2 = {
   prosecutorAppealAnnouncement: null,
   judge: null,
   notifications: null,
+  requestedDefenderName: 'rassi',
```
Comment (oddsson) [inline apps/judicial-system/web/src/utils/mocks.ts:120]: Sama komment og fyrir ofan ☝️🙏

File: apps/judicial-system/web/src/utils/mocks.ts [hunk 7]
```diff
@@ -159,6 +161,8 @@ const testCase3 = {
...
```
Comment (oddsson) [inline apps/judicial-system/web/src/utils/mocks.ts:164]: Sama komment og fyrir ofan ☝️🙏

**Review by addi — DISMISSED** [review]
Still LGTM

**Review by addi — APPROVED** [review]

**Pull request merged**